### ISSUE TITLE
Delete image doesn't delete vdisk image

### ISSUE THREAD

**Issue by zaibon** [zero-os/0-orchestrator#1117]
Version : master

## How to reproduce:
- import an image
- call the delete image API
- try to import a new image with the same name

## expected result
second image imported

## actual result
API says an image with this name already exists

**Comment by muhamadazmy** [comment]
@zaibon 
I was able to reproduce this issue and noted the following:
- It's the `zeroctl import` command what it fails, because the desk was not deleted from ardb
- ays service was gone
- it failed with me on the `second` try. The first delete went fine, and was able to import again, it's the second one that fails
- While it was failing, i ran the correct `zeroctl delete` command manually, on ays next trial to import it immediately worked.
- I think the 2nd `zeroctl delete` command was never executed, i tend to believe it's ays tendency to not run the same action again!

**Comment by zaibon** [comment]
> I think the 2nd zeroctl delete command was never executed, i tend to believe it's ays tendency to not run the same action again! 

That just mean that the orchestrator API doesn't sent a proper blueprint. It needs to force the action.

**Comment by muhamadazmy** [comment]
No it does, i might have jumped to the ays conclusion. It's not AYS 
When i tried to run the bp manually (the one generated by orchestrator) it actually ran it. But i noticed another failure in the node logs.

It's like something is wrong with the container itself, the weird thing is if i tried to do the same with the direct client  like

```python
cl.container.client(cl.container.create('https://hub.gig.tech/zaibon/0-disk-tlog-accept-address.flist', host_network=True).get()).system('/bin/zeroctl delete vdisks ovc-ubuntu-1604 --config 172.17.0.255:2379').get()
```

In one shot, it succeeds every single time !! it's only when it's running via ays it fails.

Anyway, i will keep looking into the issue, will keep u uptodate.

**Comment by muhamadazmy** [comment]
@zaibon 
I tried the same scenario multiple times trying to figure out what happens that causes this and those are my conclusions:
- It's always the 2nd time you try to delete/import that fails. On a fresh env, i was able to import, delete, then import successfully, but fails when i try to delete again. (given that no other unrelated errors happens like ardb crash)
- I noticed that the faulty delete shows this on the node logs
https://gist.github.com/muhamadazmy/5c6adaa03a6cc43ada22f0507de5a6e5
- While zeroctl complains that `no given vdisk could be found to be deleted ` the next import fails because of the vdisk exists!
- Note, while the zeroctl delete fails, the delete action itself does not fail. I think it was done this way so deleting a disk that doesn't exist doesn't return an error.

**Comment by muhamadazmy** [comment]
So after more investigation i found that the second delete actually fails silently on the node (vdisk fails to delete the disk) with the following error

```
STDERR:
t=2017-10-12T08:38:34+0000 lvl=eror msg="key 'ovc-ubuntu-1604:vdisk:conf:static' was not found on the ETCD server" module=global caller=source_etcd.go:63
t=2017-10-12T08:38:34+0000 lvl=eror msg="received invalid etcd config 'ovc-ubuntu-1604:vdisk:conf:static' (vdisk:'')" module=global caller=source_etcd.go:151
t=2017-10-12T08:38:34+0000 lvl=eror msg="no static vdisk config could be retrieved for ovc-ubuntu-1604: config is not available in source" module=global caller=vdisks.go:144
Error: no given vdisk could be found to be deleted
Usage:
  zeroctl delete vdisks vdiskid... [flags]

Flags:
      --config SourceConfig   config resource: dialstrings (etcd cluster) or path (yaml file) (default config.yml)
  -h, --help                  help for vdisks

Global Flags:
  -v, --verbose   log available information

no given vdisk could be found to be deleted

DATA:
{"status":403,"subject":"etcd","data":{"endpoints":["172.17.0.255:2379"],"key":"ovc-ubuntu-1604:vdisk:conf:static"}}

```

I checked that `etcd` key `ovc-ubuntu-1604:vdisk:conf:static` and found out it doesn't have a value. May be the 2nd import never set that key. I am still looking into it. The question is why trying to import again after that doesn't succeed !

**Comment by zaibon** [comment]
>  The question is why trying to import again after that doesn't succeed !

You can import another image with the same name. It will fail for the same reason even is the image is different

**Comment by muhamadazmy** [comment]
Yeah, that's because the disk name is somehow still in ardb, i checked 0-disk code and it checks for disk existence here https://github.com/zero-os/0-Disk/blob/master/zeroctl/cmd/backup/import.go#L145

Unlike what i expected, it checks on other keys not only the '<name>: vdisk:conf:static` in etcd, it also checks on something else in ardb.
Fortunately, this check can be overridden with the `--force` flag when importing. Which is gonna be my fix for this issue. (another better fix would be fixing `zeroctl delete vdisk` to do a proper delete by removing all the traces of the disk from ardb.

If u agree on using the `--force` flag, i can commit and push the fix now to `disaster_recovery` branch.

**Comment by muhamadazmy** [comment]
Fixed by using `--force` flag with the vdisk import
